src/NCronJob/Registry/RuntimeJobRegistry.cs

```diff
@@ -0,0 +1,233 @@
+        this.concurrencySettings = concurrencySettings;
+    }
+
+    /// <inheritdoc />
+    public void AddJob(Action<NCronJobOptionBuilder> jobBuilder)
+    {
+        var runtimeCollection = new RuntimeServiceCollection();
+        var builder = new NCronJobOptionBuilder(runtimeCollection, concurrencySettings);
+        jobBuilder(builder);
+        builder.RegisterJobs();
+
+        foreach (var jobDefinition in runtimeCollection.GetJobDefinitions())
+        {
+            jobRegistry.Add(jobDefinition);
+            jobWorker.ScheduleJob(jobDefinition);
+        }
+
+        foreach (var entry in runtimeCollection.GetDynamicJobFactoryRegistries())
+        {
+            dynamicJobFactoryRegistry.Register(entry);
+        }
+    }
+
+    /// <inheritdoc />
+    public void RemoveJob(string jobName) => jobWorker.RemoveJobByName(jobName);
+
+    /// <inheritdoc />
+    public void RemoveJob<TJob>() where TJob : IJob => RemoveJob(typeof(TJob));
+
+    /// <inheritdoc />
+    public void RemoveJob(Type type) => jobWorker.RemoveJobType(type);
+
+    /// <inheritdoc />
+    public void UpdateSchedule(string jobName, string cronExpression, TimeZoneInfo? timeZoneInfo = null)
+    {
+        ArgumentNullException.ThrowIfNull(jobName);
+        ArgumentNullException.ThrowIfNull(cronExpression);
+
+        var precisionRequired = JobOptionBuilder.DetermineAndValidatePrecision(cronExpression, null);
+        var job = jobRegistry.FindJobDefinition(jobName) ?? throw new InvalidOperationException($"Job with name '{jobName}' not found.");
+
+        var cron = NCronJobOptionBuilder.GetCronExpression(cronExpression, precisionRequired);
+
+        job.CronExpression = cron;
+        job.TimeZone = timeZoneInfo ?? TimeZoneInfo.Utc;
+
+        jobWorker.RescheduleJobByName(job);
+    }
+
+    /// <inheritdoc />
+    public void UpdateParameter(string jobName, object? parameter)
+    {
+        ArgumentNullException.ThrowIfNull(jobName);
+
+        var job = jobRegistry.FindJobDefinition(jobName) ?? throw new InvalidOperationException($"Job with name '{jobName}' not found.");
+        job.Parameter = parameter;
+
+        jobWorker.RescheduleJobByName(job);
+    }
+
+    /// <inheritdoc />
+    public bool TryGetSchedule(string jobName, out string? cronExpression, out TimeZoneInfo? timeZoneInfo)
+    {
+        cronExpression = null;
+        timeZoneInfo = null;
+
+        var job = jobRegistry.FindJobDefinition(jobName);
+        if (job is null)
+        {
+            return false;
+        }
+
+        cronExpression = job.UserDefinedCronExpression;
+        timeZoneInfo = job.TimeZone;
+
+        return true;
+    }
+
+    /// <inheritdoc />
+    public IReadOnlyCollection<RecurringJobSchedule> GetAllRecurringJobs()
+        => jobRegistry
+            .GetAllCronJobs()
+            .Select(s => new RecurringJobSchedule(
+                s.Type == typeof(DynamicJobFactory) ? null : s.Type,
+                s.CustomName,
```

Comment:
Maybe it is just a naming issue - but our JobName is not unique (and we don't give any guarantee).
Given a job with two schedules would destroy this already.

The Custom Name is a user-specified one. I don't want to hand out internals to the user. The user defines everything inside `RecurringJobSchedule`. 

Furthermore exposing our JobName would bound us to keep it stable. I am not sure if we want this.

Reply:
The JobName is more like a Tag or Category.
They aren't necessarily unique - but the whole runtime part is currently built on top of a unique jobname.

The alternative would be to hand out an internal ID when defining jobs.
But as this happens inside the configuration step of the application, it might get messy for users to retrieve those entries.

Reply:
Fair point. I do think there is value in being able to grab some type of handle back, like the ID. However as of now there is no functionality that would benefit that so could leave this as-is

Reply:
Then let's remove the `JobType` for now - reintroducing should be no problem and gives us time to rethink the approach if necessary.